debug_webserver.py

```diff
@@ -30,7 +31,15 @@ def log_message(self, format: str, *args: Any) -> None:
     def do_POST(self) -> None:  # noqa: N802 # I cant change it
         """Handle POST."""
         logger.info(f"Received a POST for {self.path!r}!")
-        data = self.rfile.read(int(self.headers['Content-Length'])).decode('utf-8', errors='replace')
+        data_raw: bytes = self.rfile.read(int(self.headers['Content-Length']))
+        data: str | bytes
+
+        if self.headers.get('Content-Encoding') == 'gzip':
```

Comment:
Although we know how the current code is sending, is there any reason to *not* also have `'deflate'` checked here as well?  EDDN Gateway code has:

```python
if content_encoding in ['gzip', 'deflate']:
```